composer/core/state.py

```diff
@@ -708,7 +708,7 @@ def train_dataloader(self, train_dataloader: Optional[Union[Iterable, DataLoader
         """
         self._train_dataloader = train_dataloader
         # Load dataset state from checkpoint when train_dataloader is set
-        if self.dataset_state:
+        if self.dataset_state is not None and self.dataset_state.get('train') is not None:
```

Comment:
Can you add a clarifying comment here why we check for `self.dataset_state.get('train') is not None`?